resources/test/fixtures/F402.py

```diff
@@ -0,0 +1,9 @@
+import os
+import os.path as path
+
+
+for os in range(3):
+    pass
+
+for path in range(3):
+    pass
```

Comment:
Flake8 has this check for `self.differentForks(node, existing.source)`, which I _think_ enables them to avoid flagging errors in cases like this:

```py
if False:
    import typing
else:
    for typing in range(3):
        pass
```

(`ruff` raises F402 here, but Flake8 doesn't.)

It might be okay to have false positives for this rule in particular, but it seems like something we'll have to solve to support `RedefinedWhileUnused` and some others. `differentForks` requires that we store the parent stack for every binding, and then implement the `differentForks` logic.

Do you think that's worth doing here, as part of this PR? Or would you rather merge w/ these limitations?

(Either way, can we add an example using the `if`/`else` to `F402.py`, even if it has a TODO saying that we shouldn't be flagging that line?)

Reply:
I guess pylint doesn't really do this, so maybe we should just not worry about it for now.

For example, pylint has a rule (`W0621`) around using a variable name that shadows a variable in the outer scope, and it does get triggered here:

```py
if True:
    typing = 1
else:
    def f(typing):
        pass
```